packages/abi/src/matchers/sway-type-matchers.test.ts

```diff
@@ -0,0 +1,48 @@
+import { swayTypeMatchers } from './sway-type-matchers';
+
+const testCases: Record<keyof typeof swayTypeMatchers, string> = {
+  empty: '()',
+  generic: 'generic T',
+  bool: 'bool',
+  u8: 'u8',
+  u16: 'u16',
+  u32: 'u32',
+  u64: 'u64',
+  u256: 'u256',
+  b256: 'b256',
+
+  string: 'str[5]',
+  array: '[_; 2]',
+  tuple: '(_, _, _)',
+
+  struct: 'struct MyStruct',
+  assetId: 'struct std::asset_id::AssetId',
+  b512: 'struct std::b512::B512',
+  bytes: 'struct std::bytes::Bytes',
+  evmAddress: 'struct std::vm::evm::evm_address::EvmAddress',
+  stdString: 'struct std::string::String',
+  vector: 'struct std::vec::Vec',
+
+  enum: 'enum MyEnum',
+  option: 'enum std::option::Option',
+  result: 'enum std::result::Result',
+
+  rawUntypedPtr: 'raw untyped ptr',
+  rawUntypedSlice: 'raw untyped slice',
+};
+
+describe('sway type matchers', () => {
+  test.each(Object.entries(testCases))('%s - %s', (type, value) => {
```

Comment:
I have given this feedback many times before, but I will repeat:
- Please do not use this sort of `test.each(...)`. As the set grows, isolating things becomes impossible, and understanding anything becomes a pain.

Reply:
https://github.com/FuelLabs/fuels-ts/pull/3137/commits/19f793df60c5e9eecfe44d13b8c2a6130da13dc9